pulser-core/pulser/sequence/sequence.py

```diff
@@ -121,6 +122,25 @@ def __init__(
                 f"'device' must be of type 'BaseDevice', not {type(device)}."
             )
 
+        with warnings.catch_warnings():
+            warnings.simplefilter("always")
```

Comment:
Why do you go into catch_warnings to set the filter on always ? (what could happen if you only had the warnings.warn ?)